src/OpenTelemetry.Exporter.Jaeger/Implimentation/JaegerUdpBatcher.cs

```diff
@@ -0,0 +1,163 @@
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Thrift.Protocols;
+
+    public class JaegerUdpBatcher : IJaegerUdpBatcher
+    {
+        private const int DefaultMaxPacketSize = 65000;
+        private readonly int? maxPacketSize;
+        private readonly ITProtocolFactory protocolFactory;
+        private readonly JaegerThriftClientTransport clientTransport;
+        private readonly JaegerThriftClient thriftClient;
+        private readonly Process process;
+        private readonly int processByteSize;
+        private readonly List<JaegerSpan> currentBatch = new List<JaegerSpan>();
+
+        private int batchByteSize;
+
+        private bool disposedValue = false; // To detect redundant calls
+
+        public JaegerUdpBatcher(JaegerExporterOptions options)
+        {
+            this.maxPacketSize = options.MaxPacketSize == 0 ? DefaultMaxPacketSize : options.MaxPacketSize;
+            this.protocolFactory = new TCompactProtocol.Factory();
+            this.clientTransport = new JaegerThriftClientTransport(options.AgentHost, options.AgentPort.Value);
+            this.thriftClient = new JaegerThriftClient(this.protocolFactory.GetProtocol(this.clientTransport));
+            this.process = new Process(options.ServiceName);
+            this.processByteSize = this.GetSize(this.process);
+            this.batchByteSize = this.processByteSize;
+        }
+
+        public async Task<int> AppendAsync(JaegerSpan span, CancellationToken cancellationToken)
+        {
+            int spanSize = this.GetSize(span);
+
+            if (spanSize > this.maxPacketSize)
+            {
+                throw new JaegerExporterException($"ThriftSender received a span that was too large, size = {spanSize}, max = {this.maxPacketSize}", null);
+            }
+
+            this.batchByteSize += spanSize;
+            if (this.batchByteSize <= this.maxPacketSize)
+            {
+                this.currentBatch.Add(span);
+
+                if (this.batchByteSize < this.maxPacketSize)
+                {
+                    return 0;
+                }
+
+                return await this.FlushAsync(cancellationToken).ConfigureAwait(false);
+            }
+
+            int n;
+
+            try
+            {
+                n = await this.FlushAsync(cancellationToken).ConfigureAwait(false);
+            }
+            catch (JaegerExporterException ex)
+            {
+                // +1 for the span not submitted in the buffer above
+                throw new JaegerExporterException(ex.Message, ex);
+            }
+
+            this.currentBatch.Add(span);
+            this.batchByteSize = this.processByteSize + spanSize;
+            return n;
+        }
+
+        public async Task<int> FlushAsync(CancellationToken cancellationToken)
+        {
+            if (this.currentBatch.Count == 0)
+            {
+                return 0;
+            }
+
+            int n = this.currentBatch.Count;
```

Comment:
could `n` be `0` at this point?
Should the value copy be done before the `if`?

Reply:
good point. not sure why the original Jaeger client put it in this order.